rest/client/sender.go

```diff
@@ -18,16 +18,23 @@ const (
 )
 
 type logSender struct {
-	logTaskData TaskData
-	logChannel  string
-	comm        Communicator
-	cancel      context.CancelFunc
-	pipe        chan message.Composer
-	lastBatch   chan struct{}
-	signalEnd   chan struct{}
+	logTaskData   TaskData
+	logChannel    string
+	comm          Communicator
+	cancel        context.CancelFunc
+	pipe          chan message.Composer
+	lastBatch     chan struct{}
+	signalEnd     chan struct{}
+	updateTimeout bool
 	*send.Base
 }
 
+func newTimeoutLogSender(ctx context.Context, comm Communicator, channel string, taskData TaskData) send.Sender {
```

Comment:
Use this new constructor in rest/client/client.go.

Reply:
done